es-app/src/views/gamelist/GridGameListView.cpp

```diff
@@ -1,16 +1,86 @@
+// Part of this was written by Aloshi but was never finished and added to ES. 
+// It would originally load all game art at ES load and could easily eat up
+// All VRAM in smaller systems, such as the raspberry pi.
+// I've changed how ImageGridComponent stores the game's art texture so
+// it will only load the images for tiles in a range of the cursor and
+// unload textures outside of its range depending on which direction the
+// user is moving.
+// ============================================================================
 
 GridGameListView::GridGameListView(Window* window, FileData* root) : ISimpleGameListView(window, root),
-	mGrid(window)
+	mGrid(window, 6), mBackgroundImage(window), mGameTitle(window),
+	mDescContainer(window), mDescription(window),
+	mImage(window),
+	
+	mLblRating(window), mLblReleaseDate(window), mLblDeveloper(window), mLblPublisher(window),
+	mLblGenre(window), mLblPlayers(window), mLblLastPlayed(window), mLblPlayCount(window),
+
+	mRating(window), mReleaseDate(window), mDeveloper(window), mPublisher(window),
+	mGenre(window), mPlayers(window), mLastPlayed(window), mPlayCount(window)
 {
-	mGrid.setPosition(0, mSize.y() * 0.2f);
-	mGrid.setSize(mSize.x(), mSize.y() * 0.8f);
+	mGameTitle.setFont(Font::get(FONT_SIZE_MEDIUM));
+	mGameTitle.setPosition(0, mSize.y() * 0.05f);
+	mGameTitle.setColor(0xAAAAAAFF);
+	mGameTitle.setSize(mSize.x(), 0);
+	mGameTitle.setAlignment(ALIGN_CENTER);
+	addChild(&mGameTitle);
+
+	mGrid.setPosition(24, mSize.y() * 0.15f);
+	mGrid.setSize(mSize.x() * 0.8f, mSize.y() * 0.8f);
+	mGrid.setMargin(Eigen::Vector2f(Renderer::getScreenWidth() * .02f, Renderer::getScreenHeight() * .05f));
 	addChild(&mGrid);
+	
+	mRoot = root;
+
+	populateList(mRoot->getChildren());
+
+	// --- DETAILED LIST METADATA ---
+	const float padding = 0.01f;
+
+	// metadata labels + values
+	mLblRating.setText("Rating: ");
+	addChild(&mLblRating);
+	addChild(&mRating);
+	mLblReleaseDate.setText("Released: ");
+	addChild(&mLblReleaseDate);
+	addChild(&mReleaseDate);
+	mLblDeveloper.setText("Developer: ");
+	addChild(&mLblDeveloper);
+	addChild(&mDeveloper);
+	mLblPublisher.setText("Publisher: ");
+	addChild(&mLblPublisher);
+	addChild(&mPublisher);
+	mLblGenre.setText("Genre: ");
+	addChild(&mLblGenre);
+	addChild(&mGenre);
+	mLblPlayers.setText("Players: ");
+	addChild(&mLblPlayers);
+	addChild(&mPlayers);
+	mLblLastPlayed.setText("Last played: ");
+	addChild(&mLblLastPlayed);
+	mLastPlayed.setDisplayMode(DateTimeComponent::DISP_RELATIVE_TO_NOW);
+	addChild(&mLastPlayed);
+	mLblPlayCount.setText("Times played: ");
+	addChild(&mLblPlayCount);
+	addChild(&mPlayCount);
 
-	populateList(root->getChildrenListToDisplay());
+	mDescContainer.setPosition(mSize.x() * padding, mSize.y() * 0.65f);
+	mDescContainer.setSize(mSize.x() * (0.50f - 2 * padding), mSize.y() - mDescContainer.getPosition().y());
+	mDescContainer.setAutoScroll(true);
+    mDescContainer.setDefaultZIndex(40);
+    addChild(&mDescContainer);
```

Comment:
wrong indentation.

Reply:
Wow thanks for noticing this, my IDE completely fucked up the indentation here for whatever reason it used 4 spaces instead of tabs.